src/themes/types.ts

```diff
@@ -10,6 +10,82 @@ import { Extendable, ObjectOf, ObjectOrFunc } from '../../types/utils'
 //
 // We use these terms in typings to indicate which phase the typings apply to.
 
+// ========================================================
+// Colors
+// ========================================================
+
+/**
+ * A type for a palette for a single color.
+ */
+export type ColorVariants = {
+  50: string
+  100: string
+  200: string
+  300: string
+  400: string
+  500: string
+  600: string
+  700: string
+  800: string
+  900: string
+}
+
+/**
+ * A type for a predefined natural colors.
+ */
+type NaturalColorsStrict = Partial<{
+  blue: ColorVariants
+  green: ColorVariants
+  grey: ColorVariants
+  orange: ColorVariants
+  pink: ColorVariants
+  purple: ColorVariants
+  teal: ColorVariants
+  red: ColorVariants
+  yellow: ColorVariants
+}>
+
+export type NaturalColors = Extendable<NaturalColorsStrict, ColorVariants>
+
+/**
+ * A type for a predefined state colors.
+ */
+export type ContextualColorsStrict = Partial<{
+  text: ColorVariants
```

Comment:
emm, not sure whether this `text` key delivers any context

Reply:
Yes, should be moved to other section somehow. We don't reached any decision there.